controllers/ratingsController.js

```diff
@@ -53,25 +63,35 @@ class RatingController {
    * @return {Object} - It returns the response object.
    */
 
-  static async getAll(req, res) {
+  static async getAll(req, res, next) {
+    const { slug } = req.params;
+    const userId = req.user.id;
     const { article } = req;
 
-    try {
-      const results = await Rating.findAll({
-        raw: true,
-        where: { articleId: article.id }
-      });
-      const averageRating = await Rating.findAll({
-        attributes: [
-          [sequelize.fn("AVG", sequelize.col("rating")), "averageRating"]
-        ],
-        raw: true,
-        where: { articleId: article.id }
-      });
-      res.status(OK).json({ ratings: results, averageRating: averageRating[0] });
-    } catch (error) {
-      res.status(INTERNAL_SERVER_ERROR).json({ message: "Please Try again later" });
-    }
+    joi.validate({ userId, slug }, ratingAll, async (err, _value) => {
+      if (err) {
+        next(err);
+      } else {
+        try {
+          const results = await Rating.findAll({
+            raw: true,
+            where: { articleId: article.id }
+          });
+          const averageRating = await Rating.findAll({
```

Comment:
I think it seems like a duplicate. You are querying twice through a single table. What do you think finding a way of adding a method on Rating model that would calculate the average?

Reply:
Looks, like a good idea but still adding it to the model I will be querying the database twice.